src/cirq_iqm/iqm_sampler.py

```diff
@@ -164,14 +173,19 @@ def _send_circuits(
             circuit_duration_check=self._circuit_duration_check,
             heralding_mode=self._heralding_mode,
         )
-
         try:
             timeout_arg = [self._run_sweep_timeout] if self._run_sweep_timeout is not None else []
             results = self._client.wait_for_results(job_id, *timeout_arg)
             if results.measurements is None:
                 raise RuntimeError('No measurements returned from IQM quantum computer.')
 
-            return [{k: np.array(v) for k, v in measurements.items()} for measurements in results.measurements]
+            return [
+                (
+                    {k: np.array(v) for k, v in measurements.items()},
+                    ResultMetadata(job_id, results.metadata.calibration_set_id, results.metadata.request),
+                )
+                for measurements in results.measurements
```

Comment:
Now each circuit in the batch gets the request for the entire batch in its metadata, right?

Reply:
Yes that is correct. `cirq` results only exist per circuit while the request is always for a batch of circuits. It's a bit confusing but I can't think of a better solution. Do you have one?

Reply:
Not really